### PR TITLE
Fix multiple removes via iterator leaving no space in the set.

### PR DESCRIPTION
**Pull request by mohrezaei**
Aligns the rehashing algorithm with the primitive maps.

### CODE REVIEW

**Comment by motlin** [comment]
I think you'll need to update the email/ip-validation but looks good to me.

**Comment by mohrezaei** [comment]
Rebased. IP validation is all good on eclipse.org

**Comment by mohrezaei** [comment]
I had to revise the rehash to fix #576

**Comment by nikhilnanivadekar** [comment]
the ip validation is still failing most likely because your commit is not signed off.

**Comment by mohrezaei** [comment]
Added signed-off.

**Comment by mohrezaei** [comment]
Fixes #551 and #576

File: unit-tests/src/test/java/org/eclipse/collections/impl/set/mutable/primitive/LongHashSetExtraTest.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        for (; putLong < 1000L; putLong++)
+        {
+            set.add(putLong);
+        }
+        for (; putLong < (1L << 30); putLong++)
+        {
+            set.add(putLong);
+            set.remove(putLong - 1000L);
+        }
+    }
```
Comment (motlin) [inline unit-tests/src/test/java/org/eclipse/collections/impl/set/mutable/primitive/LongHashSetExtraTest.java]: Does this test take a long time? If so, we should probably move it to the acceptance-tests directory.

Any ideas for a test that could check for a similar problem without iterating so many times?

Comment (motlin) [reply unit-tests/src/test/java/org/eclipse/collections/impl/set/mutable/primitive/LongHashSetExtraTest.java]: I checked the build logs and these tests take 105s.

02:26:44:287 [INFO] Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 105.513 s - in org.eclipse.collections.impl.set.mutable.primitive.LongHashSetExtraTest

File: unit-tests/src/test/java/org/eclipse/collections/impl/set/mutable/primitive/LongHashSetExtraTest.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+            set.add(putLong);
+        }
+        for (; putLong < 10_000_000L; putLong++)
+        {
+            set.add(putLong);
+            set.remove(putLong - 1000L);
+        }
+        Field table = LongHashSet.class.getDeclaredField("table");
+        table.setAccessible(true);
+        Assert.assertTrue(((long[]) table.get(set)).length < 10_000);
```
Comment (motlin) [inline unit-tests/src/test/java/org/eclipse/collections/impl/set/mutable/primitive/LongHashSetExtraTest.java]: 👍

**Pull request merged**